### PR TITLE
Fix multiple bugs in redundantMemberwiseInit rule

### PR DESCRIPTION
**Pull request by miguel-jimenez-0529**
## Summary
- Fix critical bug in the existing `redundantMemberwiseInit` rule that prevented processing multiple structs in a single file
- Fix formatting bug where first property would be incorrectly positioned after init removal
- Remove problematic `return` statement that caused early exit after processing first struct
- Add comprehensive test coverage and improve rule implementation

## Root Cause
The upstream `redundantMemberwiseInit` rule (added in #2098) contained multiple bugs:

### Bug 1: Early Return Preventing Multiple Struct Processing
```swift
// Remove redundant init if all assignments match
if isRedundant, assignmentCount == propertiesWithoutDefaults.count {
    // ... removal logic ...
    formatter.removeTokens(in: actualStartIndex ... actualEndIndex)
    return  // <-- This caused early exit after first struct
}
```

### Bug 2: Incorrect First Property Positioning  
The whitespace handling logic was too aggressive, causing the first property to be positioned incorrectly next to the struct's opening brace after init removal.

### 1. Critical Bug Fixes
- **Removed the problematic `return` statement** after `formatter.removeTokens()`  
- This allows the rule to continue processing all structs in a file during a single pass
- **Enhanced whitespace handling** to preserve proper formatting of the first property
- More conservative trailing whitespace cleanup with better bounds checking

### 2. Comprehensive Test Suite  
- Added 52+ comprehensive test cases covering all edge cases
- **Includes multi-struct test scenarios** that would have failed with the original bug
- **`testRemoveRedundantPublicMemberwiseInitWithProperFormattingOfFirstProperty`** specifically tests the first property positioning fix
- Tests complex struct hierarchies, access levels, default values, and formatting edge cases
- Added `testRemoveRedundantMemberwiseInitWithComplexStruct` with 10+ structs in one file

### 3. Implementation Improvements
- Enhanced whitespace handling during init removal to prevent formatting issues
- Improved bounds checking for safer token manipulation (`actualEndIndex + 1 < formatter.tokens.count`)
- Better handling of multi-line initializers and property declarations  
- More conservative trailing whitespace cleanup (`break` after including one newline)

## Test Coverage
All 52+ test cases pass, including critical scenarios:
- `testRemoveRedundantMemberwiseInitWithComplexStruct` - Tests 10+ structs in single file (multi-struct bug)
- `testRemoveRedundantPublicMemberwiseInitWithProperFormattingOfFirstProperty` - Tests first property positioning (formatting bug)
- Basic memberwise init removal across various struct types
- Complex access level scenarios (public/internal/private combinations)
- Properties with default values and optional parameters
- Documentation comment and formatting preservation

## Impact
- **Fixes the core functionality** - rule can now process multiple structs in a single pass
- **Fixes formatting issues** - first property maintains proper spacing after init removal
- **Enables stable output** - no more infinite formatting cycles on multi-struct files  
- **No behavioral changes** - only fixes bugs preventing proper operation
- **Production ready** - comprehensive test coverage ensures reliability

### Multi-struct Processing Bug
**Before**: In files with multiple structs, only one struct would be processed per formatting pass, requiring multiple runs to achieve stable output
**After**: All structs in a file are processed in a single pass, achieving immediate stable output

### First Property Formatting Bug  
**Before**: 
```swift
public struct CardViewAnimationState {public let style: CardStyle  // Incorrectly positioned
    public let backgroundColor: UIColor?
}
```

**After**:
```swift
public struct CardViewAnimationState {
    public let style: CardStyle  // Properly formatted
    public let backgroundColor: UIColor?
}
```

These fixes are essential for the rule to work correctly in real-world codebases where multiple structs per file are common and proper formatting is critical.

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: Sources/Rules/RedundantMemberwiseInit.swift [hunk 1]
```diff
@@ -236,25 +237,39 @@ public extension FormatRule {
...
 
-                    // Find the range including preceding and trailing whitespace
+                    // Find the range including preceding whitespace, but be conservative about trailing
                     var actualStartIndex = startRemovalIndex
                     var actualEndIndex = endRemovalIndex
 
                     // Include preceding spaces and blank line
-                    while let prevToken = formatter.token(at: actualStartIndex - 1), prevToken.isSpace {
-                        actualStartIndex -= 1
+                    while actualStartIndex > 0 {
```
Comment (calda) [inline Sources/Rules/RedundantMemberwiseInit.swift:245]: What are these other changes for?

Comment (miguel-jimenez-0529) [reply Sources/Rules/RedundantMemberwiseInit.swift:245]: image

To fix this issue

Comment (calda) [reply Sources/Rules/RedundantMemberwiseInit.swift:245]: ah makes sense

**Review by calda — APPROVED** [review]
thankss
